plugins/inventory/nb_inventory.py

```diff
@@ -599,11 +599,19 @@ def extract_interfaces(self, host):
 
             interfaces = list(interfaces_lookup[host["id"]].values())
 
+            before_netbox_v29 = bool(self.ipaddresses_lookup)
             # Attach IP Addresses to their interface
             for interface in interfaces:
-                interface["ip_addresses"] = list(
-                    self.ipaddresses_lookup[interface["id"]].values()
-                )
+                if before_netbox_v29:
```

Comment:
Do we even need to to determine if it's < 2.9?

Could we just have the two lookups, device and VM ip address lookups? So we handle the difference in versions in the code below by determining which bucket to throw them into?

Reply:
If you look in the changes I did in `refresh_ipaddresses` then you see that for NetBox v2.9 and later I will know if it is an IP address is connected to a VM or device by looking at the `assigned_object_type`. But in earlier NetBox versions this information is not available like that to determine which bucket to throw them into. Or what kind of change do you suggest? 

It would have been great if I just could have the the two lookups and skip the `ipaddresses_lookup` completely but as said I do not see that I have the information available in earlier version to put IPs in the two buckets.

Reply:
Ok I do see what you're saying now. And we can't act off anything else with the way the inventory is setup to differentiate.

I will approve and merge this in.